File: Backend/Backend/DTOs/Booking/Booking.cs
namespace Backend.DTOs.Booking
{
    public class BookingDto
    {
        public int Id { get; set; }
        public string BookingCode { get; set; } = string.Empty;
        public int CustomerId { get; set; }
        public string? CustomerName { get; set; }
        public int ServiceId { get; set; }
        public string? ServiceName { get; set; }
        public int StaffId { get; set; }
        public string? StaffName { get; set; }
        public DateTime StartTime { get; set; }
        public DateTime EndTime { get; set; }
        public string Status { get; set; } = string.Empty;
        public string? CustomerNote { get; set; }
        public string? CancellationReason { get; set; }
        public DateTime CreatedAt { get; set; }
    }
}

File: Fontend/service-booking/src/types/booking.ts
export type BookingStatus = "Pending" | "Confirmed" | "Completed" | "Cancelled";

export interface BookingItem {
  id: number;
  bookingCode: string;
  customerId: number;
  customerName: string;
  serviceId: number;
  serviceName: string;
  staffId: number;
  staffName: string;
  startTime: string;
  endTime: string;
  status: BookingStatus;
  customerNote?: string | null;
  cancellationReason?: string | null;
  createdAt: string;
}

export interface PaginatedBookingsResponse {
  data: BookingItem[];
  pageNumber: number;
  pageSize: number;
  totalCount: number;
  totalPages: number;
}

export interface CreateBookingPayload {
  serviceId: number;
  staffId: number;
  startTime: string; // YYYY-MM-DDTHH:mm:ss
  customerNote?: string;
}

export interface AvailableSlotsResponse {
  date: string;
  serviceId: number;
  staffId: number;
  availableSlots: string[];
}

export interface CancelBookingPayload {
  cancellationReason: string;
}
